src/ucp/api/ucp.h

```diff
@@ -3292,6 +3292,51 @@ ucp_atomic_fetch_nb(ucp_ep_h ep, ucp_atomic_fetch_op_t opcode,
                     ucp_send_callback_t cb);
 
 
+/**
+ * @ingroup UCP_COMM
+ * @brief Post an atomic fetch operation.
+ *
+ * This routine will post an atomic fetch operation to remote memory.
+ * The remote value is described by the combination of the remote
+ * memory address @a remote_addr and the @ref ucp_rkey_h "remote memory handle"
+ * @a rkey.
+ * The routine is non-blocking and therefore returns immediately. However the
+ * actual atomic operation may be delayed. The atomic operation is not considered complete
+ * until the values in remote and local memory are completed.
+ *
+ * @note The user should not modify any part of the @a buffer or @a result after
+ *       this operation is called, until the operation completes.
+ * @note Only contig(32) or contig(64) datatypes are supported in
+ *       @a param->datatype, see @ref ucp_dt_make_contig
+ *
+ * @param [in] ep          UCP endpoint.
+ * @param [in] opcode      One of @ref ucp_atomic_fetch_op_t.
+ * @param [in] buffer      Address of operand for atomic operation. In the
+ *                         case of CSWAP this is the conditional for the swap.
+ *                         For SWAP this is the value to be placed in remote memory.
+ * @param [inout] result   Local memory address to store resulting fetch to.
+ *                         In the case of CSWAP the value in result will be
+ *                         swapped into the @a remote_addr if the condition
+ *                         is true.
+ * @param [in] count       Number of elements in @a buffer and @a result
```

Comment:
`Number of elements in @a buffer and @a result. The size of each element is specified by @ref ucp_request_param_t.datatype`

Reply:
fixed